apps/ima/behaviours/aggregator-save-update.js

```diff
@@ -0,0 +1,176 @@
+
+module.exports = superclass => class extends superclass {
+  constructor(options) {
+    if (!options.aggregateTo) {
+      throw new Error('options.aggregateTo is required for loops');
+    }
+    if (!options.aggregateFrom) {
+      throw new Error('options.aggregateField is required for loops');
+    }
+    super(options);
+  }
+
+  deleteItem(req, res) {
+    const id = req.params.id;
+
+    let items = '';
+
+    if (id) {
+      items = this.getAggregateArray(req).filter((element, index) => index !== parseInt(id, 10));
+      this.setAggregateArray(req, items);
+    }
+
+    if(items.length === 0 && `${req.form.options.route}` === '/human-rights-family-summary') {
+      res.redirect(`${req.baseUrl}/human-rights-claim`);
+    }
```

Comment:
This is for redirecting user to the first screen when the user deletes an item from the summary table and there are no items left in the table, as mentioned in the Figma designs:

<img width="571" alt="image" src="https://github.com/UKHomeOffice/ima/assets/142597294/6354ab17-031d-42a3-9bac-f5f4a045e9de">

Reply:
Ah ok, this is a bit different from the standard looping flow we have in other services (usually it would go back to '/human-rights-family' when no others existed)  so I would say move it to a separate behaviour.

Reply:
Actually, it might work to leave it in but you would need some unit tests to test it won't interfere with other loops

Reply:
According to the `req.form.options.route` check, it should only run that logic on the `human-rights-family-summary` page, so it shouldn't interfere with other loops. From testing, the route value is only 'human-rights-family-summary' when the user presses 'Continue' on that page.

Reply:
unit tests have been added as suggested